packages/roosterjs-editor-plugins/lib/plugins/CustomReplace/CustomReplace.ts

```diff
@@ -107,7 +112,7 @@ export default class CustomReplacePlugin implements EditorPlugin {
 
         if (
             replacement.shouldReplace &&
-            !replacement.shouldReplace(replacement, searcher.getWordBefore())
+            !replacement.shouldReplace(replacement, searcher.getWordBefore(), sourceTag)
```

Comment:
And passing in a sourceTag is not general enough. What if next time we have an requirement to check something else, such as if a given element has a special attribute or style but not a tag name? We should not keep adding new parameters here. Each new parameter should be general enough.

Instead passing in the tag name, you can pass in the whole editor (as IEditor) reference then you can do any check inside the callback.

Reply:
I thought passing the whole editor would be too extreme because I do not know how big it can get, It certainly would make for the most complete replacement rules.

Reply:
It will just pass a reference (in 32bit system, just 4 bytes). So no matter how big editor is, the cost is same.

Reply:
Ok, I will pass the editor